### PR TITLE
Upgrade mysqld_exporter to v0.15.0 

### LINKED ISSUE #3169: MySQL integration fixed upstream but not in the agent
Running 0.31 version of the grafana-agent with mysql_exporter integration.

When `- info_schema.innodb_tablespaces` collector is enabled and MySQL8.0+ is scraped , the following error is logged

`Mar  3 10:00:05 mysql1 grafana-agent[40911]: ts=2023-03-03T10:00:05.978202054Z caller=exporter.go:180 level=error integration=mysqld_exporter msg="Error from scraper" scraper=info_schema.innodb_tablespaces err="Error 1109: Unknown table 'INNODB_SYS_TABLESPACES' in information_schema"
`

The table was renamed from 'INNODB_SYS_TABLESPACES' to 'INNODB_TABLESPACES' in MySQL 8.0

This was reported in https://github.com/prometheus/mysqld_exporter/issues/370 and fixed upstream here https://github.com/prometheus/mysqld_exporter/pull/516

### LINKED ISSUE #4748: Update github.com/prometheus/mysqld_exporter to v0.15.0
An update for `github.com/prometheus/mysqld_exporter` (version `v0.15.0`) is now available. Version `v0.13.0` is currently in use.

### PR DESCRIPTION
**Pull request by marctc**
#### PR Description

Bump `mysqld_exporter` to  v0.15.0. This version includes new flag for the collector
`perf_schema.memory_events`. It also has some breaking changes regarding renamed
and removed metrics.

#### Which issue(s) this PR fixes

Fixes #3169 
Fixes #4748 

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [x] Tests updated
- [ ] Config converters updated

### CODE REVIEW

File: component/prometheus/exporter/mysql/mysql.go [hunk 1]
```diff
@@ -144,6 +154,7 @@ func (a *Arguments) Convert() *mysqld_exporter.Config {
 		PerfSchemaEventsStatementsTextLimit:  a.PerfSchemaEventsStatements.TextLimit,
 		PerfSchemaFileInstancesFilter:        a.PerfSchemaFileInstances.Filter,
 		PerfSchemaFileInstancesRemovePrefix:  a.PerfSchemaFileInstances.RemovePrefix,
+		PerfSchemaMemoryEventsRemovePrefix:   a.PerfSchemaMemoryEvents.RemovePrefix,
```
Comment (erikbaranowski) [inline component/prometheus/exporter/mysql/mysql.go:157]: The opposite of this should be added to `converter/internal/staticconvert/internal/build/mysqld_exporter.go` `func toMysqldExporter`

**Review by spartan0x117 — APPROVED** [review]
Aside from the converter stuff, LGTM!

**Review by clayton-cornell — COMMENTED** [review]
Doc changes are minimal. OK as-is

**Review by clayton-cornell — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,24 @@ internal API changes are not present.
...
+  - Removed metrics:
+    - "mysql_last_scrape_failed" (gauge)
+    - "mysql_exporter_scrapes_total" (counter)
+    - "mysql_exporter_scrape_errors_total" (counter)
+  - Metric names in the `info_schema.processlist` collector have been [changed](https://github.com/prometheus/mysqld_exporter/pull/603).
+  - Metric names in the `info_schema.replica_host` collector have been [changed](https://github.com/prometheus/mysqld_exporter/pull/496).
+  - Changes related to `replication_group_member_stats collector`:
+    - metric "transaction_in_queue" was Counter instead of Gauge
+    - renamed 3 metrics starting with `mysql_perf_schema_transaction_` to start with `mysql_perf_schema_transactions_` to be consistent with column names.
+    - exposing only server's own stats by matching `MEMBER_ID` with `@@server_uuid` resulting "member_id" label to be dropped.
```
Comment (Dasomeone) [inline CHANGELOG.md:25]: Had a look through the mixin used in the integration, and there are no breaking changes.

The only change affecting the mixin is re: 
> Metric names in the `info_schema.processlist` collector have been [changed](https://github.com/prometheus/mysqld_exporter/pull/603).

However, the mixin was already using this change, as it is from 2022.

**Pull request merged**